torch_geometric/nn/models/lightgcn.py

```diff
@@ -0,0 +1,301 @@
+        alpha (float, optional): The vector specifying the layer weights in the
+            final embeddings. If set to :obj:`None`, the original paper's
+            uniform initialization of :obj:`(1 / num_layers + 1)` is used
+            (default: :obj:`None`).
+        **kwargs (optional): Additional arguments of the underlying
+            :class:`torch_geometric.nn.conv.MessagePassing` layers.
+    """
+
+    objectives = ['ranking', 'link_prediction']
+
+    def __init__(self, num_layers: int, num_nodes: int,
+                 embedding_dim: int = 32, objective: str = 'ranking',
+                 lambda_reg: float = 1e-4, alpha: Tensor = None, **kwargs):
+        super().__init__(**kwargs)
+
+        assert (num_layers >= 1), 'Number of layers must be larger than 0.'
+        assert (num_nodes >= 1), 'Number of nodes must be larger than 0.'
+        assert (embedding_dim >= 1), 'Embedding dimension must be larger ' \
+            'than 0.'
+        assert (objective in LightGCN.objectives), 'Objective must be one ' \
+            f'of {LightGCN.objectives}'
+
+        if alpha is not None:
+            assert (alpha.size(0) == (num_layers + 1)), 'alpha must be of ' \
+                'size (num_layers + 1).'
+        else:
+            alpha = [1 / (num_layers + 1)] * (num_layers + 1)
+
+        self.num_layers = num_layers
+        self.num_nodes = num_nodes
+        self.embedding_dim = embedding_dim
+        self.embeddings = Embedding(num_nodes, embedding_dim)
+        self.objective = objective
+        self.lambda_reg = lambda_reg
+        self.alpha = alpha
+        self.convs = ModuleList()
+
+        for _ in range(num_layers):
+            self.convs.append(LGConv())
+
+        if objective == LightGCN.objectives[0]:
+            self.loss_fn = BPRLoss(self.lambda_reg)
+        else:
+            self.loss_fn = BCELoss()
+
+    def reset_parameters(self):
+        self.embeddings.reset_parameters()
+
+    def forward(self, edge_index: Adj, edge_label_index: Tensor,
+                weights: Tensor = None, **kwargs) -> Tensor:
+        """Computes rankings or link probabilities for pairs of nodes.
+
+        Args:
+            edge_index (Adj): Matrix specifying the start and end nodes
+                to be predicted.
+            edge_label_index (Tensor): Tensor of edges to get rankings
+                or probabilities for.
+            weights (Tensor, optional): Pre-computed embeddings or node
+                features which should be used to initialize the embedding
+                layer (default: :obj:`None`).
+        """
+        if weights is not None:
+            assert (weights.shape == self.embeddings.shape), \
+                'Pre-computed embeddings weights must match shape of \
+                embedding layer.'
+            self.embeddings.weight.data.copy_(weights)
+
+        out = self._propagate_embeddings(edge_index)
+
+        nodes_fst = torch.index_select(out, 0, edge_label_index[0, :].long())
```

Comment:
In the case of bipartite graph where all edges are edge_label_index, it seems that if we set edge_label_index=edge_index when no edge_label_index is passed in, we can achieve by default the behaviour of the original LightGCN implementation - please feel free to correct me here if I misunderstood the implementation. Perhaps adding a default value for edge_label_index is more intuitive for users who are familiar with the original paper but not so much with the feature added by this PR.

Reply:
Bipartite default option is implemented in latest commit :) Thank you.

Reply:
Prefer
```suggestion
        out_src = out[edge_label_index[0]]
```